pkg/build/generator/generator.go

```diff
@@ -233,6 +234,9 @@ func (g *BuildGenerator) Instantiate(ctx kapi.Context, request *buildapi.BuildRe
 
 	newBuild, err := g.generateBuildFromConfig(ctx, bc, request.Revision, request.Binary)
 	if err != nil {
+		if oerrors.IsKnownAPIErrorType(err) {
```

Comment:
The correct test is casting to APIStatus in the kube `pkg/api/errors` package - if the error implements that, we consider it already handled.